databricks/koalas/indexes/datetimes.py

```diff
@@ -634,6 +636,55 @@ def strftime(self, date_format: str) -> Index:
         """
         return Index(self.to_series().dt.strftime(date_format))
 
+    def indexer_between_time(
+        self,
+        start_time: Union[datetime.time, str],
+        end_time: Union[datetime.time, str],
+        include_start: bool = True,
+        include_end: bool = True,
+    ) -> Index:
+        """
+        Return index locations of values between particular times of day
+        (e.g., 9:00-9:30AM).
+
+        Parameters
+        ----------
+        start_time, end_time : datetime.time, str
+            Time passed either as object (datetime.time) or as string in
+            appropriate format ("%H:%M", "%H%M", "%I:%M%p", "%I%M%p",
+            "%H:%M:%S", "%H%M%S", "%I:%M:%S%p","%I%M%S%p").
+        include_start : bool, default True
+        include_end : bool, default True
+
+        Returns
+        -------
+        values_between_time : Index of integers
+
+        Examples
+        --------
+        >>> kidx = ks.date_range("2000-01-01", periods=3, freq="T")
+        >>> kidx
+        DatetimeIndex(['2000-01-01 00:00:00', '2000-01-01 00:01:00',
+                       '2000-01-01 00:02:00'],
+                      dtype='datetime64[ns]', freq=None)
+
+        >>> kidx.indexer_between_time("00:01", "00:02")
+        Int64Index([1, 2], dtype='int64')
+
+        >>> kidx.indexer_between_time("00:01", "00:02", include_end=False)
+        Int64Index([1], dtype='int64')
+
+        >>> kidx.indexer_between_time("00:01", "00:02", include_start=False)
+        Int64Index([2], dtype='int64')
+        """
+
+        def pandas_between_time(pdf):
+            return pd.DataFrame(pdf.between_time(start_time, end_time, include_start, include_end))
+
+        kdf = self.to_frame().reset_index(drop=True).reset_index().set_index(DEFAULT_SERIES_NAME)
+        res_kdf = kdf.apply_batch(pandas_between_time)
```

Comment:
`kdf.koalas.apply_batch(pandas_between_time)`? `kdf.apply_batch(pandas_between_time)` is deprecated.

Reply:
Modified!